python/docs/dlis/errors.rst

```diff
@@ -0,0 +1,179 @@
+
+This error might indicate:
+
+* File is missing a chunk of data in the middle
+* File is completely zeroed-out from certain point
+* File contains a zeroed-out chunk of data
+* File is a non-standard TIF file
+
+too short logical record
+^^^^^^^^^^^^^^^^^^^^^^^^
+.. code-block:: text
+
+  Too short logical record. Length can't be less than 4, but was 0
+
+This error might indicate:
+
+* File is completely zeroed-out from certain point
+* File contains a zeroed-out chunk of data
+
+unexpected EOF when reading / file truncated
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. code-block:: text
+
+  rp66: unexpected EOF when reading record - got 1234 bytes, expected
+  there to be 5678 more
+
+.. code-block:: text
+
+  File truncated ...
+
+This error might indicate:
+
+* File is incomplete, unknown amount of information is missing from the
+  end
+
+logical record segment expects successor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: text
+
+  End of logical file, but last logical record segment expects successor
+
+.. code-block:: text
+
+  Reached EOF, but last logical record segment expects successor
+
+This error might indicate:
+
+* Logical file is incomplete, end of file or new logical file follows.
+
+Parsing errors
+--------------
+
+Some errors which occur during parsing of objects or curves.
+
+error parsing / unexpected end-of-record
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. code-block:: text
+
+  error parsing object descriptor
+
+.. code-block:: text
+
+  unexpected end-of-record
+
+.. code-block:: text
+
+  unexpected end-of-record in template
+
+These (and similar) errors indicate that file breaks specification in a way that
+is too ambiguous. When encountered, dlisio no longer knows how to interpret data
+that follows, so it stops processing data further.
+
+Some common situations that might lead to this error:
+
+* Incorrect invariant attribute in template
+* Invalid LRS padbytes value
+* Fileheader set is declared to be named, but is not
+
+field occurs more than once
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. code-block:: text
+
+  field 'DEPTH.1.0' occurs more than once
+
+Happens when accessing curves. According to specification, every channel may
+appear in the frame once and only once.
+So this error may indicate:
+
+* Something is broken in frame-channel relationships and you should be cautious
+  with trusting the data
+* Channel is repeated for user convenience. It's likely the case if repeated
+  channel can be assumed to be index of the frame, like for example "DEPTH"
+
+To bypass this error call::
+
+  frame.curves(strict=False)
+
+fmtstr would read past end
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. code-block:: text
+
+  corrupted record: fmtstr would read past end
+
+Error happens when reading curves. It indicates that more curves data is
+expected than there is actually present. dlisio doesn't know what precisely
+caused data to be incorrect, hence can't recover from it.
```

Comment:
> It indicates that more curves data is expected than there is actually present. dlisio doesn't know what precisely caused data to be incorrect, hence can't recover from it.

Or: 

There is less data than what the metadata claims there to be. Either the metadata is incorrect or the file is missing data, or both. It's not obvious which one it is. The inconsistency between the metadata and the raw data makes it impossible for dlisio to correctly parse the data. Hence there is no recovery from this.